views/ViewsData.js

```diff
@@ -153,7 +167,7 @@ export function DataProvider(props) {
 
     try {
       dispatch({ type: IS_SUBMITTING, value: true })
-      let res = await onSubmit.current(stateRef.current, args)
+      let res = await onSubmit.current(stateRef.current, args, _onChange)
```

Comment:
Nice one @alex-vladut! This should work just fine. Before we merge though, I was thinking, should we consider changing its signature to an object so we can extend it going forward more easily if needed? Something like `{ value, args, onChange }` a codemod should be pretty simple to implement to replace all of the instances of it in the apps. What do you think?

Reply:
Makes sense to me. I will make that change and we can have a chat as well about how to apply that codemod. Thanks

Reply:
Cool! Thanks!

You can see uses of `onSubmit` with:
```
findjs "function onSubmit(next"
```

For most cases, I think the codemod could be as simple as:
```
findjsr "function onSubmit(next)" "function onSubmit({ value: next })"
```

There is one on DDC that will need a special case with args:
```
  async function onSubmit(next, shouldSignAgreement = true) {
admin/src/App/MyAccount/Content/Profile/Content/MyCompanies/AddItem/Content/logic.js
```